logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb

```diff
@@ -175,28 +173,29 @@ def initialize(queue, size, wait)
         @size = size
         @wait = wait
 
-        @originals = Hash.new
-
         # TODO: disabled for https://github.com/elastic/logstash/issues/6055 - will have to properly refactor
         # @cancelled = Hash.new
 
+        @originals = java.util.HashSet.new(size * 4 / 3 + 1, 0.75)
```

Comment:
@jordansissel :D the `0.75` as a redundant default was my plan for a comment here ... a bad plan in hindsight lol
Will add a proper comment :)